### PR TITLE
Feature/podaac 5326: fix "add matching granules to download"

### PR DESCRIPTION
**Pull request by jbyrne6**
Changes:
- Added function to fetch all granules and get their granules names when clicking on "add matching to download"
- Added granule names to be sent to harmony call

Tested:
- tested and working locally and in SIT

### CODE REVIEW

**Review by sliu008 — CHANGES_REQUESTED** [review]
can you revert the src/hitideConfig.js to the normal one.

**Comment by jbyrne6** [comment]
> can you revert the src/hitideConfig.js to the normal one.

yes, I just changed it

**Review by sliu008 — APPROVED** [review]

**Comment by jamesfwood** [comment]
@jbyrne6 So for all your PRs, please remember to update the CHANGELOG.md with your ticket and details.  Thanks!

File: src/jpl/dijit/GranulesController.js [hunk 1]
```diff
@@ -620,17 +621,132 @@ define([
...
+                // Unsubscribe from cancel requests
+                topicHandler.remove();
+                console.log("ERROR", err);
+                // Set searching flag to false
+                _context._granuleSearchInProgress = false;
+
+                // Show spinner 
+                _context.displayLoadingSpinner(false);
+
+                // new AlertDialog({
```
Comment (frankinspace) [inline src/jpl/dijit/GranulesController.js:740]: Remove the commented code

File: src/jpl/dijit/GranulesController.js [hunk 2]
```diff
@@ -620,17 +621,132 @@ define([
...
+                            return granuleObj["meta"]["native-id"];
+                        });
+                    } else {
+                        granuleNamesToReturn = response.response.docs.map(function(granuleObj) {
+                            return granuleObj['Granule-Name']
+                        })
+                    }
+                    var startDate = moment.utc(_context.startDateWidget.getValue());
+                    var endDate = moment.utc(_context.endDateWidget.getValue());
+                    var queryId = Math.floor(Math.random() * (9999999999 - 0)) + 0;
```
Comment (frankinspace) [inline src/jpl/dijit/GranulesController.js:708]: What's going on here? Is this like a javascript thing to get an integer from math.random?

**Review by frankinspace — APPROVED** [review]

File: src/jpl/dijit/GranulesController.js (hunk 2 above)
Comment (jbyrne6) [reply src/jpl/dijit/GranulesController.js:708]: I am not sure what this is doing exactly. I just moved this code to a different part of the file.

**Pull request merged**